helpers/setup_json_generator.py

```diff
@@ -0,0 +1,297 @@
+
+        for key, value in optional_params:
+            if key in bool_options.keys() and isinstance(value, bool):
+                bool_options[key] = value
+            elif key in int_options.keys() and isinstance(value, int):
+                int_options[key] = value
+            elif key in str_options.keys() and isinstance(value, str):
+                str_options[key] = value
+
+        ci = {
+            "setup": bool_options['setup'],  # Required
+            "validation": bool_options['validation'],  # Not even used
+            "scenarios": bool_options['scenarios'],  # Optional
+            "cleanup": bool_options['cleanup'],  # optional
+            "send_to_testrail": bool_options['send_to_testrail'],  # Optional
+            "fail_on_failed_scenario": bool_options['fail_on_failed_scenario'],  # Optional
+            "setup_retries": int_options['setup_retries'],  # Optional
+            "scenario_retries": int_options['scenario_retries'],  # Optional
+            "version": str_options['version'],
+            "local_hypervisor": {  # enkel voor testen, niet voor setup
+                "type": "KVM",
+                "user": "root",
+                "password": "rooter"
+            },
+
+            "hypervisor": {},
+            "config_manager": str_options['config_manager'],
+            "user": {
+                "shell": {
+                    "username": "root",
+                    "password": "rooter"
+                },
+                "api": {
+                    "username": "admin",
+                    "password": "admin"
+                }
+            },
+            "grid_ip": grid_ip
+        }
+        self.json['ci'] = ci
+
+    def _add_hypervisor(self, machine_ip, vms=None):
+        if vms is None:
+            vms = {'vm_ip': {'name': '_name', 'role': '_role'}}
+        hypervisor_dict = {
+            "type": "KVM",
+            'ip': machine_ip,
+            'user': 'root',
+            'password': 'rooter',
+            'vms': vms
+        }
+        for i in vms.keys():
+            self._ips.append(i)
+        self.json['ci']['hypervisor'] = hypervisor_dict
+
+    def add_storagerouter(self, storagerouter_ip, hostname):
+        try:
+            socket.inet_aton(storagerouter_ip)
+        except TypeError:
+            raise ValueError('Invalid ip adress provided: {0}'.format(storagerouter_ip))
+        if not isinstance(hostname,str):
+            raise ValueError('Invalid hostname provided: {0} must be a string, not {1}'.format(hostname,type(hostname)))
+        sr_dict = {}
+        sr_dict[storagerouter_ip] = {"hostname": hostname,
+                                     "domains": {},
+                                     "disks": {},
+                                     "vpools": {}
+                                     }
+        self.json['setup']['storagerouters'].update(sr_dict)
+
+    def _add_disk_to_sr(self, storagerouter_ip, name, roles):
```

Comment:
Shouldn't this be made public too?
Strange name for a function which only allows to add roles to a disk
Add validation to the roles being configured

Reply:
for adding validation, I import the roleslist straight from dal, thus creating a dependency, is this allowed?